mirrord/agent/src/sniffer.rs

```diff
@@ -20,27 +23,56 @@ use rawsocket::RawCapture;
 use tokio::{
     net::UdpSocket,
     select,
-    sync::mpsc::{self, Receiver, Sender},
+    sync::{
+        broadcast,
+        mpsc::{error::TrySendError, Receiver, Sender},
+    },
 };
 use tokio_util::sync::CancellationToken;
-use tracing::{debug, error, trace, warn};
+use tracing::Level;
 
+use self::messages::{SniffedConnection, SnifferCommand, SnifferCommandInner};
 use crate::{
     error::AgentError,
     http::HttpVersion,
-    util::{ClientId, IndexAllocator, Subscriptions},
-    watched_task::TaskStatus,
+    util::{ClientId, Subscriptions},
 };
 
+pub mod api;
+pub mod messages;
```

Comment:
nit: with just `pub` we're not warned on unused, so if you need this everywhere in the agent change to `pub(crate)` please.